### PR TITLE
Cleanup for add-on documentation

### PR DESCRIPTION
**Pull request by kcoleman731**
*Description of changes:*
This pull request contains general cleanup for our add-on documentation. It fixes bugs and brings consistency to code samples. It also fixes typos in docs. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by askulkarni2 — APPROVED** [review]
:shipit:

File: docs/addons/argo-cd.md [hunk 1]
```diff
@@ -25,9 +27,18 @@ new EksBlueprint(app, 'my-stack-name', addOns, [], {
...
+
+1. Creates the namespace specified in the construction parameter (`argocd` by default).
+2. Deploys the [`argo-cd`](https://argoproj.github.io/argo-helm) Helm chart into the cluster.
+3. Allows to specify `ApplicationRepository` selecting the required authentication method as SSH Key, username/password or username/token. Credentials are expected to be set in AWS Secrets Manager and replicated to the desired region. If bootstrap repository is specified, creates the initial bootstrap application which may be leveraged to bootstrap workloads and/or other add-ons through GitOps.
+4. Allows setting the initial admin password through AWS Secrets Manager, replicating to the desired region. 
+
+## Setting an Admin Password
 
-## Setting Admin Password from AWS Secrets Manager
+By default, the Argo CD add-on will create a new admin password for you. To specify you own, you can leverage the AWS Secrets manage integration.
```
Comment (shapirov103) [inline docs/addons/argo-cd.md:41]: minor: you own = your own

Comment (shapirov103) [reply docs/addons/argo-cd.md:41]: minor: leverage the AWS Secrets manage integrations = (maybe) leverage the AWS Secrets Manager to provide the secret.

File: docs/addons/argo-cd.md [hunk 2]
```diff
@@ -126,10 +136,11 @@ The framework provides support to supply repository and administrator secrets in
...
-4. Allows setting the initial admin password through AWS Secrets Manager, replicating to the desired region. 
+## Known Issues
+
+1. Destruction of the cluster with provisioned applications may cause cloud formation to get stuck on deleting ArgoCD namespace. This happens because the server component that handles Application CRD resource is destroyed before it has a chance to clean up application that were provisioned through GitOps (of which CFN is unaware). To address this issue at the moment, App of Apps application should be destroyed manually before destroying the stack. 
+2. Changing the administrator password in the AWS Secrets Manager and rerunning the stack causes login error on ArgoCD UI. This happens due to the fact that Argo Helm rewrites the the secret containing the Dex server API Key (OIDC component of ArgoCD). The workaround at present is to restart the `argocd-server` pod, which repopulates the token. Secret management aspect of ArgoCD will be improved in the future to not require this step after password change.
```
Comment (shapirov103) [inline docs/addons/argo-cd.md:143]: minor: the the

File: docs/addons/argo-cd.md [hunk 3]
```diff
@@ -126,10 +136,11 @@ The framework provides support to supply repository and administrator secrets in
...
-## Functionality
 
-1. Creates the namespace specified in the construction parameter (`argocd` by default).
-2. Deploys the [`argo-cd`](https://argoproj.github.io/argo-helm) Helm chart into the cluster.
-3. Allows to specify `ApplicationRepository` selecting the required authentication method as SSH Key, username/password or username/token. Credentials are expected to be set in AWS Secrets Manager and replicated to the desired region. If bootstrap repository is specified, creates the initial bootstrap application which may be leveraged to bootstrap workloads and/or other add-ons through GitOps.
-4. Allows setting the initial admin password through AWS Secrets Manager, replicating to the desired region. 
+## Known Issues
+
+1. Destruction of the cluster with provisioned applications may cause cloud formation to get stuck on deleting ArgoCD namespace. This happens because the server component that handles Application CRD resource is destroyed before it has a chance to clean up application that were provisioned through GitOps (of which CFN is unaware). To address this issue at the moment, App of Apps application should be destroyed manually before destroying the stack.
```
Comment (shapirov103) [inline docs/addons/argo-cd.md:142]: minor: application = applications (I know it was just carried over)

File: docs/addons/container-insights.md [hunk 4]
```diff
@@ -43,12 +49,17 @@ daemonset.apps/cloudwatch-agent     1         1         1       1            1
...
 
-To enable or disable control plane logs with the console, please run the following command in your terminal - `aws eks update-cluster-config \
+To enable or disable control plane logs with the console, run the following command in your terminal.
+
+```bash
+aws eks update-cluster-config \
     --region us-east-2 \
     --name east-dev \
-    --logging '{"clusterLogging":[{"types":["api","audit","authenticator","controllerManager","scheduler"],"enabled":true}]}'`
+    --logging '{"clusterLogging":[{"types":["api","audit","authenticator","controllerManager","scheduler"],"enabled":true}]}'
```
Comment (shapirov103) [inline docs/addons/container-insights.md:58]: what happened to the PR from Aus team claiming they have this implemented with a custom resource?

File: docs/addons/external-dns.md [hunk 5]
```diff
@@ -9,11 +9,10 @@ The add-on provides functionality to configure IAM policies and Kubernetes servi
 ```typescript
 import * as ssp from '@shapirov/cdk-eks-blueprint';
 
-readonly externalDns = new ssp.addons.ExternalDnsAddon({
-    hostedZone: new ssp.addons.LookupHostedZoneProvider(myHostedZoneName)
-});
-
-const addOns: Array<ClusterAddOn> = [ externalDns ];
+const hostedZoneName '...'
```
Comment (shapirov103) [inline docs/addons/external-dns.md:12]: minor: `hostedZoneName '...'` - on purpose to have a compile issue? why not `hostedZoneName = ...` or something similar.

**Review by shapirov103 — COMMENTED** [review]
Looks good, minor comments, mostly carry-over.

File: docs/addons/container-insights.md (hunk 4 above)
Comment (kcoleman731) [reply docs/addons/container-insights.md:58]: not sure. No PRs from them yet...

**Review by shapirov103 — APPROVED** [review]
lgtm

**Pull request merged**